receiver/podmanreceiver/receiver_test.go

```diff
@@ -74,14 +68,15 @@ func TestScraperLoop(t *testing.T) {
 		}
 	}()
 
-	assert.NoError(t, r.Start(ctx, componenttest.NewNopHost()))
+	assert.NoError(t, r.start(ctx, componenttest.NewNopHost()))
 
-	md := <-consumer
+	md, err := r.scrape(ctx)
+	assert.NoError(t, err)
 	assert.Equal(t, 1, md.ResourceMetrics().Len())
 
 	assertStatsEqualToMetrics(t, genContainerStats(), md)
 
-	assert.NoError(t, r.Shutdown(ctx))
+	assert.NoError(t, r.shutdown(ctx))
```

Comment:
It might be slightly more clear if we defer this assertion and shutdown call right after we start it successfully. That way start and shutdown are grouped together, and we don't have to look further to make sure it's being called 👍

Reply:
Makes sense to me, https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32981/commits/2bf6a4a07f474eba20969e6a925f932e5f840cb1